### PR TITLE
Bugfix: scan can go into an infinite loop when used with multi-az cluster

### PR DESCRIPTION
**Pull request by jcacciatore**
The bug fix isn't with the cursor handling directly, it's with other contributing factors such as the retry policy (if `RetryNTimes` is used), not taking any action when the local rack is not picked up, not being able to set the count as a scan parameter, etc.

This PR also modifies the demo test app. I split out the properties it expects into a file and some enforcement. Also added the ability to toggle between netflix prod + test for discovering dynomite clusters.

### CODE REVIEW

File: dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java [hunk 1]
```diff
@@ -712,13 +712,13 @@ private String getCursorValue(final ConnectionContext state, final CursorBasedRe
         return "0";
     }
 
-    private List<OperationResult<ScanResult<String>>> scatterGatherScan(final CursorBasedResult<String> cursor, final String... pattern) {
+    private List<OperationResult<ScanResult<String>>> scatterGatherScan(final CursorBasedResult<String> cursor, final int count, final String... pattern) {
```
Comment (ipapapa) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java]: So we are adding `count` although it was not there before? Are we adding a new API call?

Comment (jcacciatore) [inline dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java]: Yes, but only for dyno-client. The `count` parameter is part of the redis api and is supported by jedis. Without this change the default of 10 is always used.

File: dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/RetryNTimes.java [hunk 2]
```diff
@@ -44,22 +45,31 @@ public void begin() {
 
 	@Override
 	public void success() {
-		count.incrementAndGet();
+		boolean success = false;
```
Comment (ipapapa) [inline dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/RetryNTimes.java]: what is the purpose of this change?

File: dyno-demo/src/main/java/com/netflix/dyno/demo/redis/DynoJedisDemo.java [hunk 3]
```diff
@@ -223,10 +224,12 @@ public void runScanTest(boolean populateKeys) throws Exception {
 
         CursorBasedResult<String> cbi = null;
         do {
-            cbi = client.dyno_scan(cbi, "DynoClientTest_key-*");
-
-            logger.info("result: " + cbi.getResult().toString());
+			cbi = client.dyno_scan(cbi, 100, "DynoClientTest_key-*");
```
Comment (ipapapa) [inline dyno-demo/src/main/java/com/netflix/dyno/demo/redis/DynoJedisDemo.java]: there are 500 keys written above, but you are doing count in 100. Is this correct?

File: dyno-demo/src/main/java/com/netflix/dyno/demo/redis/DynoJedisDemo.java [hunk 4]
```diff
@@ -562,36 +565,39 @@ private String constructRandomValue(int sizeInKB) {
...
-		String url = "http://discovery.cloudqa.netflix.net:7001/discovery/v2/apps/" + clusterName;
+
+		String env = System.getProperty("netflix.environment", "test");
+		String discoveryKey = String.format("dyno.demo.discovery.%s", env);
+
+		if (!System.getProperties().containsKey(discoveryKey)) {
+			throw new IllegalArgumentException("Discovery URL not found");
+		}
+
+		final String url = String.format("http://%s/%s", System.getProperty(discoveryKey), clusterName);
```
Comment (ipapapa) [inline dyno-demo/src/main/java/com/netflix/dyno/demo/redis/DynoJedisDemo.java]: Interesting, you are constructing the URL based on the properties... Looks good!

**Comment by ipapapa** [comment]
LGTM!
